File: app/src/main/java/com/asuka/androidopensourcelibrarystudydemo/view/RetrofitActivity.java
package com.asuka.androidopensourcelibrarystudydemo.view;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.asuka.androidopensourcelibrarystudydemo.R;
import com.asuka.androidopensourcelibrarystudydemo.databinding.ActivityRetrofitBinding;
import com.asuka.androidopensourcelibrarystudydemo.modle.api.Api;
import com.asuka.androidopensourcelibrarystudydemo.modle.api.WanAndroidApi;
import com.asuka.androidopensourcelibrarystudydemo.modle.pojo.BaseResponse;
import com.asuka.androidopensourcelibrarystudydemo.utils.HTTP.MyCallBack;
import com.asuka.androidopensourcelibrarystudydemo.utils.HTTP.RetrofitFactory;

import java.io.IOException;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import timber.log.Timber;

public class RetrofitActivity extends AppCompatActivity {

    public ActivityRetrofitBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityRetrofitBinding.inflate(getLayoutInflater());
//        setContentView(R.layout.activity_retrofit);
        setContentView(binding.getRoot());
        Api api = RetrofitFactory.getClient().create(Api.class);

        binding.getBtn.setOnClickListener(view -> {
            api.get("aaa","bbb").enqueue(new Callback<ResponseBody>() {
                @Override
                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                    try {
                        Timber.d(response.body().string());
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                }

                @Override
                public void onFailure(Call<ResponseBody> call, Throwable t) {
                    Timber.d(t);
                }
            });
        });

        binding.postBtn.setOnClickListener(view -> {
            api.post("123","456").enqueue(new MyCallBack<ResponseBody>() {
                @Override
                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                }
            });
        });

        binding.postWanBtn.setOnClickListener(view -> {
            Retrofit wanAndroidClient = RetrofitFactory.getWanAndroidClient();
            WanAndroidApi wanAndroidApi = wanAndroidClient.create(WanAndroidApi.class);
            wanAndroidApi.login("Fujimiya_Asuka","1593572580").enqueue(new MyCallBack<BaseResponse>() {
                @Override
                public void onResponse(Call<BaseResponse> call, Response<BaseResponse> response) {
                    Timber.d(response.body().toString());
                }
            });
        });

//        binding.postBtn.setOnClickListener(view -> api.post("123","456").enqueue(new Callback<ResponseBody>() {
//            @Override
//            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
//                try {
//                    Timber.d(response.body().string());
//                } catch (IOException e) {
//                    e.printStackTrace();
//                }
//            }
//
//            @Override
//            public void onFailure(Call<ResponseBody> call, Throwable t) {
//                Timber.d(t);
//            }
//        }));

    }
}

File: app/src/main/java/com/asuka/androidopensourcelibrarystudydemo/view/activity/RetrofitActivity.kt
package com.asuka.androidopensourcelibrarystudydemo.view.activity

import com.asuka.androidopensourcelibrarystudydemo.databinding.ActivityRetrofitBinding
import com.asuka.androidopensourcelibrarystudydemo.modle.api.HttpBinApi
import com.asuka.androidopensourcelibrarystudydemo.modle.api.WApi
import com.asuka.http.Interceptor.ConsumerInterceptor
import com.asuka.http.Interceptor.log.HttpLogInterceptor
import com.asuka.http.RXHttp
import com.asuka.http.response.DialogResponse
import com.asuka.http.utils.NetWorkScheduler
import okhttp3.ResponseBody
import timber.log.Timber

class RetrofitActivity : BaseActivity<ActivityRetrofitBinding>() {
    override fun initViewBinding(): ActivityRetrofitBinding {
        return ActivityRetrofitBinding.inflate(layoutInflater)
    }

    override fun initView() {

    }

    override fun initListener() {

        binding.postWanBtn.setOnClickListener {
            RXHttp.Instance.custom(WApi::class.java).login("Fujimiya_Asuka","1593572580")
                .compose(NetWorkScheduler().ioMain())
                .subscribe(object : DialogResponse<ResponseBody>(this){
                    override fun onSuccess(data: ResponseBody) {
                        Timber.e(data.string())
                    }
                    override fun onFailure(e: Throwable) {
                        Timber.e(e)
                    }
                })
        }

        binding.temporaryReqBtn.setOnClickListener{
            Timber.d("temp")
            RXHttp.Instance.custom("https://httpbin.org/get/",HttpBinApi::class.java).getRequest()
                .compose(NetWorkScheduler().ioMain())
                .subscribe(object : DialogResponse<ResponseBody>(this){
                    override fun onSuccess(body: ResponseBody) {
                        Timber.d(body.string())
                    }

                    override fun onFailure(e: Throwable) {
                        Timber.d(e)
                    }

                })
        }

    }

    override fun initData() {
        RXHttp.Instance.baseUrl="https://www.wanandroid.com"
        RXHttp.Instance.addInterceptor(HttpLogInterceptor())
        RXHttp.Instance.addInterceptor(ConsumerInterceptor())
    }

}